wagtail/contrib/settings/models.py

```diff
@@ -100,25 +106,36 @@ def get_page_url(self, attribute_name, request=None):
         self._page_url_cache[attribute_name] = url
         return url
 
+    @classmethod
+    def _get_locale(cls, *args, **kwargs):
+        return None
+
     def __getstate__(self):
         # Ignore 'page_url' when pickling
         state = super().__getstate__()
         state.pop("page_url", None)
         return state
 
 
-class BaseSiteSetting(AbstractSetting):
-    site = models.OneToOneField(
-        Site,
-        unique=True,
-        db_index=True,
-        editable=False,
-        on_delete=models.CASCADE,
-    )
-
+class AbstractSiteSetting(AbstractSetting):
```

Comment:
I do think there are too many classes here now. Can we not keep this as `BaseSiteSetting`, and subclass that to define `BaseTranslatableSiteSetting`? The less prohibitive `site` field on `BaseTranslatableSiteSetting`  should take precedence when extended to create a concrete model.